x-pack/platform/test/onechat_api_integration/utils/llm_proxy/scenarios.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { last } from 'lodash';
+import type { ToolingLog } from '@kbn/tooling-log';
+import type { LlmProxy } from '.';
+import { toolCallMock } from './mocks';
+
+export async function setupEsqlScenario(
+  llmProxy: LlmProxy,
+  log: ToolingLog,
+  options: {
+    userPrompt: string;
+    title: string;
+    finalLlmResponse: string;
+    esqlQuery: string;
+  }
+) {
```

Comment:
This is why I have concerns with this approach for complex scenario: this setup code is going to break every time we change anything in the internals of the search tool, and I'm afraid it will become more and more difficult to keep up with all the internal calls.

But let's see where it leads us.

Reply:
I actually debated deleting this `setupEsqlScenario` entirely, or moving it to the file where it's used. Like you, I am not sure it's reusable in its current form. However, the underlying interceptors/mocks definitely are. As more tests are added I think patterns will emerge making it easier to see how we can make helpers to mock multiple requests with a single command, instead of specifying each separately.